packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2101,7 +2097,13 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
      * @param fullGC - true to bypass optimizations and force full generation of GC data.
      */
     public async getGCData(fullGC?: boolean): Promise<IGarbageCollectionData> {
-        return this.dataStores.getGCData(fullGC);
+        const builder = new GCDataBuilder();
```

Comment:
This doesn't look sustainable long term. What if someone writes something they want GCed, but we don't call getGCData for it? For example the atMentions instance. We may want to eventually allow GC to GC it we're going to need a more robust way. I'm thinking something like an interface every GC able object has to inherit.

Reply:
That's a good point. We already have plans to move GC related functionality into GC specific interface that only objects participating in GC would implement - https://github.com/microsoft/FluidFramework/issues/8578.
With this in place, BlobManager would also implement this interface and we will know it participates in GC.

Reply:
Ok, so for example, someone implements a new type of GCable object, how are they going to get it GCed? Are we going to specifically look for the interface, or will they have to add it to the container?

Reply:
It depends on which layer the object is at. Each layer decides what is GC'able. Container runtime for example, decides that data stores and blobs can be GC'd. It then calls into data stores which then decides what object it has are GC'able and so on.
In the model that I am currently thinking of, we will query objects for the interface and if it implements that, we run GC on it. That's how most of the things in FF work today.